src/mono/mono/mini/mini-runtime.c

```diff
@@ -4408,6 +4408,31 @@ mini_llvm_init (void)
 #endif
 }
 
+#ifdef ENSURE_PRIMARY_STACK_SIZE
+/*++ 
+ Function: 
+   EnsureStackSize 
+  
+ Abstract: 
+   This fixes a problem on MUSL where the initial stack size reported by the 
+   pthread_attr_getstack is about 128kB, but this limit is not fixed and 
+   the stack can grow dynamically. The problem is that it makes the 
+   functions ReflectionInvocation::[Try]EnsureSufficientExecutionStack 
+   to fail for real life scenarios like e.g. compilation of corefx. 
+   Since there is no real fixed limit for the stack, the code below 
+   ensures moving the stack limit to a value that makes reasonable 
+   real life scenarios work. 
+  
+ --*/
+static MONO_NO_OPTIMIZATION MONO_NEVER_INLINE void
+ensure_stack_size (void)
+{
+	const int default_size = 5 * 1024 * 1024;
+	volatile uint8_t *s = (uint8_t *)g_alloca(default_size);
```

Comment:
nit: in coreclr pal (and nativeaot), we pass the size as argument from call site. It would be nice to keep it aligned:

```c
ensure_stack_size (size_t size)
{
	volatile uint8_t *s = (uint8_t *)g_alloca(size);
```

and call it with `ensure_stack_size(1536 * 1024)`.

Reply:
Pushed the requested change, thanks :)